### PR TITLE
GDScript: Use "is" keyword for type checking.

### PR DESCRIPTION
**Pull request by 27thLiz**
Replaces the `extends` keyword with `is` in the context of testing for type compatibility.
`extends` is still used for declaring class inheritance.

Example:

```gdscript
extends Node2D

func _input(ev):
    if ev is InputEventKey:
	print("yay, key event")
```

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

**Comment by Geequlim** [comment]
Add it to 2.1 please

**Comment by reduz** [comment]
probably would be a better idea if this is just for 3.0, so we can break
compat

File: modules/gdscript/gd_tokenizer.h [hunk 1]
```diff
@@ -96,6 +96,7 @@ class GDTokenizer {
 		TK_PR_FUNCTION,
 		TK_PR_CLASS,
 		TK_PR_EXTENDS,
+		TK_PR_IS,
```
Comment (bojidar-bg) [inline modules/gdscript/gd_tokenizer.h]: You forgot to update the token names on `modules/gdscript/gd_tokenizer.cpp:92`

**Review by bojidar-bg — DISMISSED** [review]
Everything else LGTM.

**Pull request merged**